src/main/java/io/cdap/plugin/dlp/Utils.java

```diff
@@ -172,18 +173,34 @@ public static StructuredRecord getStructuredRecordFromTable(Table table, Structu
   }
 
   /**
-   * Creates a DLP Table from a StructuredRecord
+   * Creates a DLP Table using the all fields from StructuredRecord
    *
    * @param record StructuredRecord to convert to Table
    * @return Table with one row containing the data from {@code record}
    * @throws Exception when record contains unexpected type or type-casting fails due to invalid values
    */
   public static Table getTableFromStructuredRecord(StructuredRecord record) throws Exception {
+    return getTableFromStructuredRecord(record, null);
+  }
+
+  /**
+   * Creates a DLP Table using the required fields from StructuredRecord
+   *
+   * @param record         StructuredRecord to convert to Table
+   * @param requiredFields Set of fields to include in the conversion, pass null to include all fields
+   * @return Table with one row containing the data from {@code record}
+   * @throws Exception when record contains unexpected type or type-casting fails due to invalid values
+   */
+  public static Table getTableFromStructuredRecord(StructuredRecord record, HashSet<String> requiredFields)
```

Comment:
Use `Set` instead of `HashSet`. Also, if it can be null, annotate it with `@Nullable`. However, I prefer not to allow null, but requires an empty set to represent nothing.

Reply:
Updated it to use empty set instead of null